### PR TITLE
WIP: Gettext scheme

### PR DESCRIPTION
**Pull request by xingxing**
This PR is mere use for design discussion. Please don't consider to merge it.

The background please check this [discussion#14840](https://github.com/renovatebot/renovate/discussions/14840)

The workflow of i18n is:
1. Extract text that are considered to be translated, generate POT (PO template) files.
2. Generate PO files from POTs, and translate `msgid` to `msgstr` on PO files.
3. Convert PO files to JSON format, so that `gettext.js` can load these JSON files.
4. Extract more POT from source code, and merge new `msgid` to existed PO files. `goto: 3` 

I wrote a Makefile to implement that process. 

Warm expecting for any feedback

### CODE REVIEW

File: locales/pot/github.po [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the renovate-platform package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: renovate-platform\n"
```
Comment (HonkingGoose) [inline locales/pot/github.po:9]: I don't think we should bother with reviewing translations in a language we don't speak at a near-native level. We can't rely on Google Translate giving us correct translations, that are fit for purpose.

Can we decouple the translations from the main English Renovate project?

The Pro Git 2 book project has a main English language repository, and different repositories for each translation. Would something like that work for us?

- [Pro Git 2 book in English](https://github.com/progit/progit2)
- [Pro Git 2 book in German](https://github.com/progit/progit2-de)
- [Pro Git organization repository overview](https://github.com/progit)

File: package.json [hunk 2]
```diff
@@ -175,6 +175,7 @@
     "find-packages": "9.0.9",
     "find-up": "5.0.0",
     "fs-extra": "10.1.0",
+    "gettext.js": "1.2.0",
```
[the remark this answers is not in the document]
Comment (HonkingGoose) [reply package.json:178]: Looks like `node-gettext` is even less well maintained as `gettext.js`.

If we can only choose between those two packages I'd take `gettext.js`.

File: package.json [hunk 3]
```diff
@@ -175,6 +175,7 @@
     "find-packages": "9.0.9",
     "find-up": "5.0.0",
     "fs-extra": "10.1.0",
```
[the remark this answers is not in the document]
Comment (HonkingGoose) [reply package.json:177]: I'll let one of the maintainers help you with upgrading third-party modules, I don't know anything about that. 😉

File: locales/pot/github.po (hunk 1 above)
Comment (HonkingGoose) [reply locales/pot/github.po:9]: The basic setup for the translation should probably live in the English language main repository, so `renovatebot/renovate`. The translation files could then go in a new repository for each language.

Each translation would then be a fork of the main `renovatebot/renovate` repository.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (xingxing) [reply package.json:178]: `node-gettext` can let us use `PO`/`MO` file directory, that provide a great benefit on development, so we need not to convert a PO file to a JSON format as we have to do by using `gettext.js`.
 
Let us give `node-gettext` a try and I will push a new PR for that soon.

[the remark this answers is not in the document]
Comment (xingxing) [reply package.json:178]: I  composed a new #17768, it seems better than this scheme.

**Pull request closed without merging**